drv/ignition-server/src/main.rs

```diff
@@ -29,6 +29,7 @@ ringbuf!(Trace, 16, Trace::None);
 
 const TIMER_NOTIFICATION_MASK: u32 = 1 << 0;
 const TIMER_INTERVAL: u64 = 1000;
+const PORT_MAX: usize = 40;
```

Comment:
Why is this 40 instead of 35? Perhaps I'm misunderstanding the purpose here or am lacking knowledge of where the other 5 Ignition devices come from. 

32 sleds
2  PSCs
1  other switch
35 total devices

Reply:
The number of controllers is actually variable, determined at RTL synthesis time and published through a register (see `port_count()` at https://github.com/oxidecomputer/hubris/blob/090ad7345c1252ce032a32980d5ffca94bddb61b/drv/sidecar-mainboard-controller/src/ignition.rs#L59). The reason for this is so I can have smaller builds for development (the emulator top has 4 controller ports for example) and scale the Hubris driver accordingly. Currently the `presence_summary` vector allows for up to 40 ports (35 rounded up to a multiple of 8). This `PORT_MAX` constant mirrors that limit and is used below to statically allocate the array used to shuttle the state dump between Ignition and the control plane and/or Humility.

I'll add a comment.